src/main.rs

```diff
@@ -1,91 +1,102 @@
-extern crate systemstat;
-use chrono::DateTime;
-use chrono::TimeZone;
-use chrono::Utc;
-use socketcan::*;
-use std::env;
-use std::io::Write;
-use std::os::unix::net::UnixStream;
-use std::process::Command;
-use std::sync::mpsc::channel;
-use std::thread;
-mod data;
-mod decode_data;
-mod master_mapping;
-mod message;
+use std::{
+    env,
+    process,
+    thread,
+    time::Duration
+};
+
+extern crate paho_mqtt as mqtt;
+
+const DFLT_BROKER:&str = "tcp://localhost:1883";
+const DFLT_CLIENT:&str = "rust_subscribe";
+const DFLT_TOPICS:&[&str] = &["/mpu", "/tpu"];
+// The qos list that match topics above.
+const DFLT_QOS:&[i32] = &[0, 1];
+
+// Reconnect to the broker when connection is lost.
+fn try_reconnect(cli: &mqtt::Client) -> bool
+{
+    println!("Connection lost. Waiting to retry connection");
+    for _ in 0..12 {
+        thread::sleep(Duration::from_millis(5000));
+        if cli.reconnect().is_ok() {
+            println!("Successfully reconnected");
+            return true;
+        }
+    }
+    println!("Unable to reconnect after several attempts.");
+    false
+}
+
+// Subscribes to multiple topics.
+fn subscribe_topics(cli: &mqtt::Client) {
+    if let Err(e) = cli.subscribe_many(DFLT_TOPICS, DFLT_QOS) {
+        println!("Error subscribes topics: {:?}", e);
+        process::exit(1);
+    }
+}
 
 fn main() {
-    let args: Vec<String> = env::args().collect();
-    let default = "tmp/ipc.sock".to_owned();
-    let ipc_path = args.get(0).unwrap_or(&default);
+    let host = env::args().nth(1).unwrap_or_else(||
```

Comment:
this is not correct. Calypso is a shared application, NERO also uses this file, so you have to figure out what application is using it from the command line args. You need to bring back the other stuff